core/src/test/java/io/questdb/cutlass/line/tcp/AlterTableAttachPartitionFromSoftLinkLineTest.java

```diff
@@ -0,0 +1,194 @@
+                }
+                qdb.getCairoEngine().releaseAllWriters();
+                qdb.getCairoEngine().releaseAllReaders();
+                try (TableReader reader = qdb.getCairoEngine().getReader(AllowAllCairoSecurityContext.INSTANCE, tableName)) {
+                    TxReader txFile = reader.getTxFile();
+                    Assert.assertNotNull(txFile);
+                    Assert.assertTrue(txFile.isPartitionReadOnly(0));
+                    Assert.assertFalse(txFile.isPartitionReadOnly(1));
+                    Assert.assertTrue(txFile.isPartitionReadOnlyByPartitionTimestamp(readOnlyPartitionTs));
+                    Assert.assertFalse(txFile.isPartitionReadOnlyByPartitionTimestamp(activePartitionTs));
+                }
+
+                // so that we know when the table writer is returned to the pool
+                final SOCountDownLatch tableWriterReturnedToPool = new SOCountDownLatch(1);
+                qdb.getCairoEngine().setPoolListener((factoryType, thread, name, event, segment, position) -> {
+                    if (Chars.equalsNc(tableName, name)) {
+                        if (factoryType == PoolListener.SRC_WRITER && event == PoolListener.EV_RETURN) {
+                            tableWriterReturnedToPool.countDown();
+                        }
+                    }
+                });
+
+                long[] timestampNano = {readOnlyPartitionTs * 1000L, activePartitionTs * 1000L};
+
+                try (LineTcpSender sender = LineTcpSender.newSender(Net.parseIPv4("127.0.0.1"), ILP_PORT, ILP_BUFFER_SIZE)) {
+                    // send O3 hitting a read-only partition
+                    for (int tick = 0; tick < 1000; tick++) {
+                        timestampNano[tick % 2] += 100_000L;
+                        sender.metric(tableName)
+                                .tag("s", "oyster")
+                                .field("l", 44)
+                                .field("i", 2023)
+                                .at(timestampNano[tick % 2]);
+                    }
+                    sender.flush();
+
+                    // send only hitting the active partition
+                    for (int tick = 0; tick < 5; tick++) {
+                        timestampNano[1] += 100_000L;
+                        sender.metric(tableName)
+                                .tag("s", "oyster")
+                                .field("l", 44)
+                                .field("i", 2023)
+                                .at(timestampNano[1]);
+                    }
+                    sender.flush();
+                }
+                tableWriterReturnedToPool.await();
+                assertSql(
+                        compiler,
+                        context,
+                        "SELECT min(ts), max(ts), count() FROM " + tableName + " sample by 1d",
+                        Misc.getThreadLocalBuilder(),
+                        "min\tmax\tcount\n" +
```

Comment:
Why there are no new records in active partition?

Reply:
i don't think this applies anymore cannot find the code